src/ASM/LR/ASMu3D.h

```diff
@@ -355,6 +382,26 @@ class ASMu3D : public ASMunstruct, public ASM3D
   virtual bool evalSolution(Matrix& sField, const IntegrandBase& integrand,
                             const RealArray* gpar, bool regular = true) const;
 
+  //! \brief Projects inhomogenuous (scalar) dirichlet conditions by continuous L2-fit.
+  //! \param[in] edge low-level edge information needed to do integration
+  //! \param[in] values inhomogenuous function which is to be fitted
+  //! \param[out] result fitted value in terms of control-point values
+  //! \param[in] time time used in dynamic problems
+  bool faceL2projection (const DirichletFace& edge,
+                         const RealFunc& values,
+                         RealArray& result,
+                         double time) const;
+
+  //! \brief Projects inhomogenuous (vector) dirichlet conditions by continuous L2-fit.
+  //! \param[in] edge low-level edge information needed to do integration
+  //! \param[in] values inhomogenuous function which is to be fitted
+  //! \param[out] result fitted value in terms of control-point values
+  //! \param[in] time time used in dynamic problems
+  bool faceL2projection (const DirichletFace& edge,
+                         const VecFunc& values,
+                         Real2DMat& result,
+                         double time) const;
+
```

Comment:
Consider merging these two almost-identical methods into one, using the `FunctionBase` common interface for the function to project.